src/pages/docs/chat/api/javascript/message.mdx

```diff
@@ -0,0 +1,393 @@
+
+`equal(message: Message): boolean`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `message` | The message to compare against. | [`Message`](#message) |
+
+### Returns
+
+Returns `true` if the two messages are the same message.
+
+<Code>
+```javascript
+const isEqual = message1.equal(message2);
+```
+</Code>
+
+## Is message the same as a given message <a id="isSameAs"/>
+
+`message.isSameAs()`
+
+Alias for [`message.equal()`](LINK).
+
+`isSameAs(message: Message): boolean`
+
+## Is message a newer version of a given message <a id="isNewerVersionOf"/>
+
+`message.isNewerVersionOf()`
+
+Determines if a message is a newer version of a given message.
+
+Note that negating this function does not mean that the message is an older version of the same message, as the two may be different messages entirely.
+
+`isNewerVersionOf(message: Message): boolean`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `message` | The message to compare against. | [`Message`](#message) |
+
+### Returns
+
+Returns `true` if the two messages are the same message (`isSameAs` returns `true`) and this message is a newer version.
+
+<Code>
+```javascript
+const isNewer = message1.isNewerVersionOf(message2);
+```
+</Code>
+
+## Is message an older version of a given message <a id="isOlderVersionOf"/>
+
+`message.isOlderVersionOf()`
+
+Determines if the message is an older version of a given message.
+
+Note that negating this function does not mean that the message is a newer version of the same message, as the two may be different messages entirely.
+
+`isOlderVersionOf(message: Message): boolean`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `message` | The message to compare against. | [`Message`](#message) |
+
+### Returns
+
+Returns `true` if the two messages are the same message (`isSameAs` returns true) and this message is an older version.
+
+<Code>
+```javascript
+const isOlder = message1.isOlderVersionOf(message2);
+```
+</Code>
+
+## Is message the same version as a given message <a id="isSameVersionAs"/>
+
+`message.isSameVersionAs()`
+
+Determines if the message is the same version as a given message.
+
+`isSameVersionAs(message: Message): boolean`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `message` | The message to compare against. | [`Message`](#message) |
+
+### Returns
+
+Returns `true` if the two messages are the same message and have the same version.
+
+<Code>
+```javascript
+const isSameVersion = message1.isSameVersionAs(message2);
+```
+</Code>
+
+## Create a copy of a message <a id="copy"/>
+
+`message.copy()`
+
+Creates a copy of the message with fields replaced per the parameters.
```

Comment:
Possibly explain the use-case here? Either for calling `update`, having temporary state in your app whilst you wait for requests to complete, etc.

Reply:
9c7b90d8a014b6e86a1508190809e427a40d7b08